File: tools/dbench/answers/go_sonnet/s3/e1_parse_fallback.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	inputs := []string{"12", "x", "7", "-", "30"}
	total := 0
	failures := 0
	for _, v := range inputs {
		if n, err := strconv.Atoi(v); err == nil {
			total += n
		} else {
			failures++
		}
	}
	fmt.Println(total)
	fmt.Println(failures)
}

File: tools/dbench/answers/rust_sonnet/s3/e1_parse_fallback.rs
fn main() {
    let inputs = ["12", "x", "7", "-", "30"];
    let mut total = 0i64;
    let mut failures = 0usize;
    for s in inputs.iter() {
        match s.parse::<i64>() {
            Ok(v) => total += v,
            Err(_) => failures += 1,
        }
    }
    println!("{}", total);
    println!("{}", failures);
}
